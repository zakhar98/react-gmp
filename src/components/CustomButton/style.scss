.button {
  cursor: pointer;
  font-size: 22px;
  border: 2px solid #f65261;
  border-radius: 4px;
  padding: 8px 24px;
  outline: none;

  &.button_contained {
    background-color: #f65261;
    color: white;
  }

  &.button_outlined {
    background-color: transparent;
    color: #f65261;
  }
}
