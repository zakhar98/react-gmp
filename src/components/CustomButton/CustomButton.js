import React from "react";
import PropTypes from 'prop-types';
import "./style.scss";

export default function CustomButton({children, outlined, ...rest}) {
    return (
      <button
        className={`button button_${outlined ? 'outlined' : 'contained'}`}
        {...rest}
        >
        {children}
      </button>
    );
}

CustomButton.propTypes = {
  children: PropTypes.node,
  outlined: PropTypes.bool,
}
