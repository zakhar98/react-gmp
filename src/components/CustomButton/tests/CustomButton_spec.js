import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from '../CustomButton.js';

describe('CustomButton', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CustomButton />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
