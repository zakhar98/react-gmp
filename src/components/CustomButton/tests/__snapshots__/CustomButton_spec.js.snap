// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders correctly 1`] = `
<button
  className="button button_contained"
/>
`;
